ui/raidboss/data/06-ew/raid/p12s.ts

```diff
@@ -508,6 +520,181 @@ const triggerSet: TriggerSet<Data> = {
+            return output.lightBeam!();
+          return output.darkBeam!();
+        }
+      },
+      outputStrings: {
+        lightBeam: {
+          en: 'light beam',
+          cn: '引导光激光',
+        },
+        darkBeam: {
+          en: 'dark beam',
+          cn: '引导暗激光',
+        },
+      },
+    },
+    {
+      id: 'P12S Engravement 1 Tower Drop Notifier',
+      type: 'GainsEffect',
+      netRegex: { effectId: ['DFB', 'DFC'] },
+      condition: (data) => data.engravementCounter === 1,
+      delaySeconds: 1, // Make sure Drop Tower Pos Tracker is done
+      alertText: (data, matches, output) => {
+        data.engravement1DarkBeamsPos = [];
+        data.engravement1LightBeamsPos = [];
+        if (data.me === matches.target) {
+          data.engravement1BeamsPosMap.forEach((value: string, key: string) => {
+            if (matches.effectId === 'DFB' && value === 'light') {
+              if (key === 'NE')
+                data.engravement1LightBeamsPos.push(output.northeast!());
+              else if (key === 'NW')
+                data.engravement1LightBeamsPos.push(output.northwest!());
+              else if (key === 'SE')
+                data.engravement1LightBeamsPos.push(output.southeast!());
+              else if (key === 'SW')
+                data.engravement1LightBeamsPos.push(output.southwest!());
+            } else if (matches.effectId === 'DFC' && value === 'dark') {
+              if (key === 'NE')
+                data.engravement1DarkBeamsPos.push(output.northeast!());
+              else if (key === 'NW')
+                data.engravement1DarkBeamsPos.push(output.northwest!());
+              else if (key === 'SE')
+                data.engravement1DarkBeamsPos.push(output.southeast!());
+              else if (key === 'SW')
+                data.engravement1DarkBeamsPos.push(output.southwest!());
+            }
+          });
+
+          // if light tower
+          if (matches.effectId === 'DFB') {
+            return output.lightTowerSide!({
+              pos1: data.engravement1LightBeamsPos[0],
+              pos2: data.engravement1LightBeamsPos[1],
+            });
+          }
+
+          return output.darkTowerSide!({
+            pos1: data.engravement1DarkBeamsPos[0],
+            pos2: data.engravement1DarkBeamsPos[1],
+          });
+        }
+      },
+      outputStrings: {
+        lightTowerSide: {
+          en: 'Drop light tower at ${pos1}/${pos2}',
+          cn: '去 ${pos1}/${pos2} 放光塔',
+        },
+        darkTowerSide: {
+          en: 'Drop dark tower at ${pos1}/${pos2}',
+          cn: '去 ${pos1}/${pos2} 放暗塔',
+        },
+        lightTower: {
+          en: 'Drop light tower',
+          cn: '放光塔',
+        },
+        darkTower: {
+          en: 'Drop dark tower',
+          cn: '放暗塔',
+        },
+        northeast: Outputs.northeast,
+        northwest: Outputs.northwest,
+        southeast: Outputs.southeast,
+        southwest: Outputs.southwest,
```

Comment:
Since there's two directions, can these be `Outputs.dirNE` instead of `Outputs.northeast` (i.e. `NE` instead of `northeast`) just to keep it shorter?

Reply:
Great, I'm sorry I didn't know this output during development.